Galaxus.Functional.Tests/Option/Async/AsyncOptionExtensions.MatchAsyncTest.cs

```diff
@@ -0,0 +1,446 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace Galaxus.Functional.Tests.Option.Async;
+
+[TestFixture]
+public class MatchAsyncTest
+{
+    [Test]
+    public async Task Option_MatchAsyncSome_Works()
+    {
+        var some = "hello".ToOption();
+
+        {
+            var called = false;
+            await some.MatchAsync(async _ =>
+            {
+                called = true;
+                await Task.CompletedTask;
```

Comment:
We've deactivated the "method is async but there's no await"-warning for these files, so the calls can be simply `await some.MatchAsync(async _ => { called = true; }, async () => { });`

My left eye always starts twitching when I see `await Task.CompletedTask` :D

Reply:
I can change these lambdas not to be async anymore and `return Task.CompletedTask`. Then we can get rid of `await Task.CompletedTask`, but still be consistent with the code where the warning is not deactivated